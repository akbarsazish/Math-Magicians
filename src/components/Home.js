import Header from './Header';

const Home = () => (
  <>
    <Header />
    <div className="homePage">
      <h2 className="welcome"> Welcome to our page! </h2>
      <p className="paragraph">
        At our website, we aim to provide you with an intuitive and user-friendly experience.
        Our home page is designed to welcome you with a modern and appealing layout,
        offering quick access to our calculator and quotes pages.
        On our calculator page, you will find a versatile and efficient tool to perform various
        mathematical calculations. complex equations for academic or professional purposes,
        our calculator has got you covered.
        For those who find inspiration in the world of mathematics,
        our quotes page is a treasure trove of wisdom and motivation.
      </p>
      <p className="paragraph">
        At our website, we aim to provide you with an intuitive and user-friendly experience.
        Our home page is designed to welcome you with a modern and appealing layout,
        offering quick access to our calculator and quotes pages.
        On our calculator page, you will find a versatile and efficient tool to perform various
        mathematical calculations. complex equations for academic or professional purposes,
        our calculator has got you covered.
        For those who find inspiration in the world of mathematics,
        our quotes page is a treasure trove of wisdom and motivation.
      </p>
    </div>
  </>
);
export default Home;
