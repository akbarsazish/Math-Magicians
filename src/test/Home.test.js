import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Home from '../components/Home';

// Mock the Header component for testing
jest.mock('../components/Header', () => {
  function Header() {
    return (
      <header role="banner">
        {/* Mocked Header content */}
      </header>
    );
  }
  return Header;
});

test('Home page displays welcome message', () => {
  render(<Home />);

  const welcomeMessage = screen.getByText('Welcome to our page!');
  expect(welcomeMessage).toMatchSnapshot();
});

test('Home page displays paragraphs', () => {
  render(<Home />);

  // Use container to query the paragraphs based on their text content
  const container = screen.getByTestId('homePageContainer');
  const paragraphs = container.querySelectorAll('p');

  expect(paragraphs).toHaveLength(2);

  expect(paragraphs[0]).toMatchSnapshot();
  expect(paragraphs[1]).toMatchSnapshot();
});

test('Home page contains the Header component', () => {
  render(
    <Router>
      <Home />
    </Router>,
  );

  const headerElement = screen.getByRole('banner');
  expect(headerElement).toMatchSnapshot();
});
