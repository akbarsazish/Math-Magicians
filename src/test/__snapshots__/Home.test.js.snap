// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page contains the Header component 1`] = `
<header
  role="banner"
/>
`;

exports[`Home page displays paragraphs 1`] = `
<p
  class="paragraph"
>
  At our website, we aim to provide you with an intuitive and user-friendly experience. Our home page is designed to welcome you with a modern and appealing layout, offering quick access to our calculator and quotes pages. On our calculator page, you will find a versatile and efficient tool to perform various mathematical calculations. complex equations for academic or professional purposes, our calculator has got you covered. For those who find inspiration in the world of mathematics, our quotes page is a treasure trove of wisdom and motivation.
</p>
`;

exports[`Home page displays paragraphs 2`] = `
<p
  class="paragraph"
>
  At our website, we aim to provide you with an intuitive and user-friendly experience. Our home page is designed to welcome you with a modern and appealing layout, offering quick access to our calculator and quotes pages. On our calculator page, you will find a versatile and efficient tool to perform various mathematical calculations. complex equations for academic or professional purposes, our calculator has got you covered. For those who find inspiration in the world of mathematics, our quotes page is a treasure trove of wisdom and motivation.
</p>
`;

exports[`Home page displays welcome message 1`] = `
<h2
  class="welcome"
>
   Welcome to our page! 
</h2>
`;
